src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -807,15 +805,9 @@ private static unsafe nuint GetIndexOfFirstNonAsciiChar_Sse2(char* pBuffer, nuin
 
             firstVector = Sse2.LoadVector128((ushort*)pBuffer); // unaligned load
 
-            // The operation below forces the 0x8000 bit of each WORD to be set iff the WORD element
-            // has value >= 0x0800 (non-ASCII). Then we'll treat the vector as a BYTE vector in order
-            // to extract the mask. Reminder: the 0x0080 bit of each WORD should be ignored.
-
-            currentMask = (uint)Sse2.MoveMask(Sse2.AddSaturate(firstVector, asciiMaskForAddSaturate).AsByte());
-
-            if ((currentMask & NonAsciiDataSeenMask) != 0)
+            if (VectorContainsNonAsciiChar(firstVector))
```

Comment:
Its a bit unfortunate that these methods are being touched as well. it will make it harder to isolate the diff/perf improvement.

I think its an overall positive change though, so I think I'm, okay with it staying part of this PR.

Reply:
I agree and IIRC I suggested to use `VectorContainsNonAsciiChar` at other places too. But I think let us revert those changes and concentrate on single method that we are trying to optimize.

Reply:
Rolled back all changes to `GetIndexOfFirstNonAsciiChar_Sse2`